pool.go

```diff
@@ -16,6 +16,19 @@ var (
 	errPoolTimeout = errors.New("redis: connection pool timeout")
 )
 
+// PoolMetrics contains pool state information and accumulated stats
+type PoolMetrics struct {
+	Conns struct {
+		Free int // the number of free connections in the pool
+		Open int // the number of open connections in the pool
+	}
+	Counts struct {
```

Comment:
`Counts` name is not very descriptive... Probably change struct to have flat layout:

```
Gets uint64
Waits uint64
Timeouts uint64

TotalConns uint64
FreeConns uint64
```